grails-app/domain/org/pih/warehouse/shipping/ShipmentItem.groovy

```diff
@@ -260,6 +278,14 @@ class ShipmentItem implements Comparable, Serializable {
         return invoiceItems?.sum { it.quantity ?: 0 } ?: 0
     }
 
+    Integer getQuantityToInvoice() {
+        return quantity - quantityInvoiced
```

Comment:
What's your plan for eventually handling cancellations/returns?

quantityInvoiced loops through all the invoices and sums the quantity, which works fine for the case where there are no quantity adjustments but... if a shipment item has quantity 3 and we cancelled or returned the shipment, we generate an invoice with quantity 0. In that scenario quantityInvoiced for the shipment item is 0, and isInvoiceable will be true (which is incorrect). We'll need some way to note that even if we invoice for less than the quantity shipped for that shipment item, we've still "fully invoiced" it.

Side note: I remember chatting with Manon about whether we need to be able to split a single shipment item up into multiple invoices (Ex: we have a shipment item with quantity 3 which has an invoice item with quantity 1 and another invoice item with quantity 2) and I thiiiiink she said we don't need the ability to do that (probably worth confirming).

So maybe it'd just be a matter of changing the getQuantityInvoiced method to return 0 for shipment items without an invoice and shipmentItem.quantity for shipment items that do have an invoice. So we ignore the quantity in the invoiceItem for the purpose of determining if the shipment item has been fully invoiced. (Which we can do because we can safely assume that each shipmentItem will only ever have a single invoiceItem so if it was invoiced at all, it's done.)

I know the work to implement cancellations isn't covered by this ticket, I just want to make sure it'll eventually be able to work within this system.